R/Sound_DMR.R

```diff
@@ -812,24 +812,31 @@ add_changepoint_info <- function(whole_df, changepoint, col_name) {
   MethRegionName <- paste0('MethRegion_', col_name)
   MethRegionLengthName <- paste0('MethRegionLength_', col_name)
   MethGroupName <- paste0('MethGroup_', col_name)
+  MethCytosinesName <- paste0('MethCytosines_', col_name)
   
   if (typeof(changepoint) != 'NULL') {
     changepoint <<- changepoint
     changepoints <- changepoint@cpts
     
     #Add in the first and last cytosine if needed for the changepoint start and stop sites
     if ((1 %in% changepoints) & (nrow(whole_df) %in% changepoints)) {
-      start <- changepoints
-      stop <- c(changepoints[-1] - 1, nrow(whole_df))
+      start <- c(1, changepoints[-length(changepoints)]+1)
+      stop <- c(changepoints[-length(changepoints)], nrow(whole_df))
+      #start <- changepoints+1
+      #stop <- c(changepoints[-1] - 1, nrow(whole_df))
     } else if ((1 %in% changepoints) & (!nrow(whole_df) %in% changepoints)) {
-      start <- changepoints
-      stop <- c(changepoints[-1] - 1, nrow(whole_df))
+      start <- c(1, changepoints[-length(changepoints)]+1)
+      stop <- c(changepoints[-length(changepoints)], nrow(whole_df))
+      #start <- changepoints
+      #stop <- c(changepoints[-1] - 1, nrow(whole_df))
     } else if ((!1 %in% changepoints) & (nrow(whole_df) %in% changepoints)) {
-      start <- c(1, changepoints[-length(changepoints)])
-      stop <- c(changepoints[-length(changepoints)] - 1, nrow(whole_df))
+      start <- c(1, changepoints[-length(changepoints)]+1)
+      stop <- c(changepoints[-length(changepoints)], nrow(whole_df))
     } else {
-      start <- changepoints
-      stop <- changepoints - 1
+      start <- c(1, changepoints[-length(changepoints)]+1)
+      stop <- c(changepoints[-length(changepoints)], nrow(whole_df))
```

Comment:
I remember there being some weird cases of start being first cytosine or end being last cytosine, which is why I had to write this awful-looking if statement block. Did you test this code with weird changepoints like that? Otherwise this whole if statement seems like it could be replaced with the following

```R
start <- c(1, changepoints[-length(changepoints)]+1)
stop <- c(changepoints[-length(changepoints)], nrow(whole_df))
```